### PR TITLE
[Merged by Bors] - feat: port Topology.UniformSpace.Completion

### PR DESCRIPTION
**Pull request by RemyDegenne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by RemyDegenne** [comment]
The only remaining errors are three wrong names in #align, due to the `private` attribute, that I don't know how to fix.

File: Mathlib/Topology/UniformSpace/Completion.lean [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+/-- Space of Cauchy filters
+
+This is essentially the completion of a uniform space. The embeddings are the neighbourhood filters.
+This space is not minimal, the separated uniform space (i.e. quotiented on the intersection of all
+entourages) is necessary for this.
+-/
+def CauchyCat (α : Type u) [UniformSpace α] : Type u :=
+  { f : Filter α // Cauchy f }
+set_option linter.uppercaseLean3 false in
+#align Cauchy CauchyCat
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Completion.lean:66]:
```suggestion
def Cauchy (α : Type u) [UniformSpace α] : Type u :=
  { f : Filter α // Cauchy f }
set_option linter.uppercaseLean3 false in
#align Cauchy Cauchy
```

File: Mathlib/Topology/UniformSpace/Completion.lean [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+
+namespace CauchyCat
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Completion.lean:68]:
```suggestion
namespace Cauchy
```

**Review by jcommelin — COMMENTED** [review]
Mathport does/did? something "silly" when a Lean 3 type starts with an uppercase letter, by appending `Cat`.

File: Mathlib/Topology/UniformSpace/Completion.lean (hunk 1 above)
Comment (RemyDegenne) [reply Mathlib/Topology/UniformSpace/Completion.lean:66]: Right, that Cat in the name made no sense. But now we have an issue : Cauchy is already taken. We need a new name.

File: Mathlib/Topology/UniformSpace/Completion.lean [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+#align uniform_space.completion.uniform_continuous_extension UniformSpace.Completion.uniformContinuous_extension
+
+theorem continuous_extension : Continuous (Completion.extension f) :=
+  cPkg.continuous_extend
+#align uniform_space.completion.continuous_extension UniformSpace.Completion.continuous_extension
+
+end CompleteSpace
+
+/- porting note: removed `@[simp]` because this lemma doesn't even trigger on itself in Lean 3 or
+Lean 4 unless the user manually supplies the `hf` argument, so it is useless as a `simp` lemma. -/
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Completion.lean:553]: Did the linter complain? Usually the linter is smart enough to pass `hf` to simp. If `simp [hf]` makes the lemma trigger on itself, then the linter should not complain.

File: Mathlib/Topology/UniformSpace/Completion.lean [hunk 4]
```diff
@@ -0,0 +1,728 @@
...
+theorem uniformContinuous_map : UniformContinuous (Completion.map f) :=
+  cPkg.uniformContinuous_map cPkg f
+#align uniform_space.completion.uniform_continuous_map UniformSpace.Completion.uniformContinuous_map
+
+theorem continuous_map : Continuous (Completion.map f) :=
+  cPkg.continuous_map cPkg f
+#align uniform_space.completion.continuous_map UniformSpace.Completion.continuous_map
+
+/- porting note: removed `@[simp]` because this lemma doesn't even trigger on itself in Lean 3 or
+Lean 4 unless the user manually supplies the `hf` argument, so it is useless as a `simp` lemma. -/
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Completion.lean:593]: Idem

File: Mathlib/Topology/UniformSpace/Completion.lean [hunk 5]
```diff
@@ -0,0 +1,728 @@
...
+protected def extension₂ (f : α → β → γ) : Completion α → Completion β → γ :=
+  cPkg.extend₂ cPkg f
+#align uniform_space.completion.extension₂ UniformSpace.Completion.extension₂
+
+section SeparatedSpace
+
+variable [SeparatedSpace γ] {f}
+
+/- porting note: removed `@[simp]` because this lemma doesn't even trigger on itself in Lean 3 or
+Lean 4 unless the user manually supplies the `hf` argument, so it is useless as a `simp` lemma. -/
```
Comment (jcommelin) [inline Mathlib/Topology/UniformSpace/Completion.lean:684]: Idem

File: Mathlib/Topology/UniformSpace/Completion.lean (hunk 3 above)
Comment (RemyDegenne) [reply Mathlib/Topology/UniformSpace/Completion.lean:553]: Yes the linter complained for all three lemmas.

**Comment by urkud** [comment]
> The only remaining errors are three wrong names in #align, due to the `private` attribute, that I don't know how to fix.

I don't `#align` private lemmas because the main purpose of `#align` is to make `mathport` use the aligned name in other files. We don't need this for private lemmas.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**